pilot/pkg/model/sidecar.go

```diff
@@ -477,15 +477,39 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 			for _, importedHost := range importedHosts {
 				// Check if the hostnames match per usual hostname matching rules
 				if importedHost.Matches(s.Hostname) {
-					// TODO: See if the service's ports match.
-					//   If there is a listener port for this Listener, then
-					//   check if the service has a port of same value.
-					//   If not, check if the service has a single port - and choose that port
-					//   if service has multiple ports none of which match the listener port, check if there is
-					//   a virtualService with match Port
-					importedServices = append(importedServices, s)
-					hostFound = true
-					break
+					portMatched := false
+					// If a listener is defined with port, we should match services with port.
+					if ilw.IstioListener != nil && ilw.IstioListener.Port != nil {
+						for _, port := range s.Ports {
+							if port.Port == int(ilw.IstioListener.Port.GetNumber()) {
+								portMatched = true
+								break
+							}
+						}
+						// If service does not have a matching port, check to see if there is a virtual service with matching port.
+						if !portMatched {
+							for _, vs := range ilw.virtualServices {
```

Comment:
nuke this logic please. I am not sure if it makes sense to look at the destination port of a virtual service. In my mind, the egress listener port (even the API says its a listener), is for listeners.